api/tests/Validator/ContentNode/AssertCompatibleWithEntityValidatorTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Tests\Validator\ContentNode;
+
+use App\Entity\Camp;
+use App\Entity\ContentNode\ColumnLayout;
+use App\Entity\ContentType;
+use App\Validator\ContentNode\AssertCompatibleWithEntity;
+use App\Validator\ContentNode\AssertCompatibleWithEntityValidator;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+/**
+ * @internal
+ */
+class AssertCompatibleWithEntityValidatorTest extends ConstraintValidatorTestCase {
+    public function testExpectsMatchingAnnotation() {
+        $this->expectException(UnexpectedTypeException::class);
+        $this->validator->validate(null, new Email());
+    }
+
+    public function testExpectsContentNodeValue() {
+        $this->expectException(UnexpectedValueException::class);
+        $this->validator->validate(new \stdClass(), new AssertCompatibleWithEntity());
+    }
+
+    public function testExpectsContentNodeObject() {
+        // given
+        $contentType = $this->createMock(ContentType::class);
+        $this->setObject(new \stdClass());
+
+        // then
+        $this->expectException(UnexpectedValueException::class);
+
+        // when
+        $this->validator->validate($contentType, new AssertCompatibleWithEntity());
+    }
+
+    public function testExpectsContentTypeValue() {
+        // given
+        $contentType = $this->createMock(Camp::class);
+        $contentNode = new ColumnLayout();
+        $this->setObject($contentNode);
+
+        // then
+        $this->expectException(UnexpectedValueException::class);
+
+        // when
+        $this->validator->validate($contentType, new AssertCompatibleWithEntity());
+    }
+
+    public function testValid() {
+        // given
+        $contentType = $this->createMock(ContentType::class);
+        $contentType->entityClass = 'App\Entity\ContentNode\ColumnLayout';
```

Comment:
```suggestion
        $contentType->entityClass = ColumnLayout::class;
```
Das wird für renaming und verschieben besser erkannt, plus ist kürzer.

Reply:
Fixed